src/battle_animation.cpp

```diff
@@ -311,6 +315,92 @@ void BattleAnimationBattle::ShakeTargets(int str, int spd, int time) {
 	}
 }
 
+BattleAnimationBattler::BattleAnimationBattler(const lcf::rpg::Animation& anim, std::vector<Game_Battler*> battlers, bool only_sound, int cutoff_frame, bool set_invert) :
+	BattleAnimation(anim, only_sound, cutoff_frame), battlers(std::move(battlers))
+{
+	invert = set_invert;
+}
+
+void BattleAnimationBattler::Draw(Bitmap& dst) {
+	if (IsOnlySound())
+		return;
+	if (animation.scope == lcf::rpg::Animation::Scope_screen) {
+		DrawAt(dst, SCREEN_TARGET_WIDTH / 2, SCREEN_TARGET_HEIGHT / 3);
+		return;
+	}
+
+	for (auto* battler: battlers) {
+		DrawAt(dst, battler->GetDisplayX(), battler->GetDisplayY());
+	}
+}
+
+void BattleAnimationBattler::UpdateScreenFlash() {
+	int r, g, b, p;
+	UpdateFlashGeneric(screen_flash_timing, r, g, b, p);
+	if (r > 0 || g > 0 || b > 0 || p > 0) Main_Data::game_screen->FlashOnce(r, g, b, p, 0);
+}
+
+void BattleAnimationBattler::FlashTargets(int r, int g, int b, int p) {
+	if (r > 0 || g > 0 || b > 0 || p > 0) {
+		for (auto& battler: battlers) {
+			battler->Flash(r, g, b, p, 0);
+		}
+	}
+}
+
+void BattleAnimationBattler::ShakeTargets(int str, int spd, int time) {
+	for (auto& battler: battlers) {
+		battler->ShakeOnce(str, spd, time);
+	}
+}
+
+void BattleAnimationBattler::DrawAt(Bitmap& dst, int x, int y) {
+	if (IsDone()) {
+		return;
+	}
+
+	const lcf::rpg::AnimationFrame& anim_frame = animation.frames[GetRealFrame()];
+
+	std::vector<lcf::rpg::AnimationCellData>::const_iterator it;
+	for (it = anim_frame.cells.begin(); it != anim_frame.cells.end(); ++it) {
+		const lcf::rpg::AnimationCellData& cell = *it;
+
+		if (!cell.valid) {
+			// Skip unused cells (they are created by deleting cells in the
+			// animation editor, resulting in gaps)
+			continue;
+		}
+
+		SetX(invert ? x - cell.x : cell.x + x);
+		SetY(cell.y + y);
+		int sx = cell.cell_id % 5;
+		if (invert) sx = 4 - sx;
+		int sy = cell.cell_id / 5;
+		int size = animation.large ? 128 : 96;
+		SetSrcRect(Rect(sx * size, sy * size, size, size));
+		SetOx(size / 2);
+		SetOy(size / 2);
+		SetTone(Tone(cell.tone_red * 128 / 100,
+			cell.tone_green * 128 / 100,
+			cell.tone_blue * 128 / 100,
+			cell.tone_gray * 128 / 100));
+		SetOpacity(255 * (100 - cell.transparency) / 100);
+		SetZoomX(cell.zoom / 100.0);
+		SetZoomY(cell.zoom / 100.0);
+		SetFlipX(invert);
+		for (auto& battler: battlers) {
+			SetFlashEffect(battler->GetFlashColor());
+		}
+		Sprite::Draw(dst);
```

Comment:
Is this copy-paste here really necessary?
You are already overloading ``FlashTargets`` can you tell us what is not working here making this necessary? Then we can work on a fix.

Reply:
The copy-pasted code is unfortunately needed for making the flash effects on battle animation sprites work.

Reply:
Removed the flash and shake stuff now ([see below](https://github.com/EasyRPG/Player/pull/2482#issuecomment-810398696)).